### ISSUE TITLE
Query combination generates wrong es query in boolean operation

### ISSUE THREAD

**Issue by wataru-chocola** [elastic/elasticsearch-dsl-py#442]
Version: 0.0.11

`elasticsearch-dsl-py` generates wrong es query in boolean operation when combining `~` and another operators.

```python
In [1]: from elasticsearch_dsl.query import Bool, QueryString
In [3]: q_a = QueryString(query='a:*')
In [4]: q_b = QueryString(query='b:*')

In [6]: str(q_a & q_b)
Out[6]: "Bool(must=[QueryString(query='a:*'), QueryString(query='b:*')])"

# wrong result, I think
In [7]: str(~(q_a & q_b))
Out[7]: "Bool(must_not=[QueryString(query='a:*'), QueryString(query='b:*')])"
```

Expected query is:

```python
Bool(must_not=[Bool(must=[QueryString(query='a:*'), QueryString(query='b:*')])])
```

**Comment by HonzaKral** [comment]
Thanks for the report @wataru-chocola, it is indeed a bug, an error on my part. Fix incoming

**Comment by HonzaKral** [comment]
So actualy `~(q1 & q2)` should be `~q1 | ~q2` which would translate to `Bool(should=[Bool(must_not=[QueryString(query='a:*') ]), Bool(must_not=[QueryString(query='b:*')])`. I implemented this logic in afab225efbc79d96b73896e64e6b84af888bdc26

Thanks again for the report!
